src/erf.jl

```diff
@@ -70,80 +70,83 @@ const sb5  =  2.55305040643316442583e+03 # 0x40A3F219, 0xCEDF3BE6
 const sb6  =  4.74528541206955367215e+02 # 0x407DA874, 0xE79FE763
 const sb7  = -2.24409524465858183362e+01 # 0xC03670E2, 0x42712D62
 
-function erfc1(x::Float64)
+function erfc1{T<:Union{Float32,Float64}}(x::T)
     s = abs(x) - 1
-    P = @horner s pa0 pa1 pa2 pa3 pa4 pa5 pa6
-    Q = @horner s 1.0 qa1 qa2 qa3 qa4 qa5 qa6
-    return 1 - erx - P/Q
+    P = @horner s T(pa0) T(pa1) T(pa2) T(pa3) T(pa4) T(pa5) T(pa6)
```

Comment:
Looking at all of these, got me wondering whether having the option of passing the type to the macro and it rewriting the type conversions would be useful.